### ISSUE TITLE
Periodically have to destroy build directory

### ISSUE THREAD

**Issue by illuzen** [trufflesuite/truffle#638]
- [*] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

Many times I will try to deploy and I get 

```
Error encountered, bailing. Network state unknown. Review successful transactions manually.
Error: The contract code couldn't be stored, please check your gas amount.
    at Object.callback (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:218485:46)
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:34886:25
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:220423:9
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:72910:11
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:204149:9
    at XMLHttpRequest.request.onreadystatechange (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:205574:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:73069:18)
    at XMLHttpRequest._setReadyState (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:73359:12)
    at XMLHttpRequest._onHttpResponseEnd (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:73514:12)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:73474:24)

```

If I then rm -r build/ and redeploy, it deploys fine. I believe this is related to

https://github.com/trufflesuite/truffle/issues/637

Operating System: OSX 10.12.6
Truffle version: Truffle v3.4.11
Ethereum client: geth 1.7.2
node version: v7.7.4
npm version: 5.5.1

**Comment by fvictorio** [comment]
Do you have that error when using a testnet or testrpc?

This may be obvious advice, but just in case: I've got that error before when using geth and a testnet because the node was not fully synced yet. Maybe in the time that it took you to remove the directory the node finished syncing? Just a wild guess.

**Comment by illuzen** [comment]
Private network, using geth 1.7.2. My node is the only node. And the problem is consistent, I have to do it every time. I've been unable to reproduce this with small contracts, so I think it's a subtle bug that only emerges beyond a certain level of code complexity. Please ask more questions to help me track this down.

**Comment by davesag** [comment]
Interesting.  I hit an issue yesterday to which the solution was also just to nuke the `build` folder and recompile. Ref https://ethereum.stackexchange.com/questions/30064/casting-from-msg-sender-address-to-a-contract-compiles-but-causes-test-failure/30154#30154

**Comment by owocki** [comment]
Is anyone able to create reproduction steps for this issue?

**Comment by illuzen** [comment]
I think it comes from creating two contracts in two different files, say A.sol and B.sol and then having an abstract contract in A.sol with the name "B". We started appending numbers to the abstract contract names so they don't collide and it seems to workaround.

**Comment by cgewecke** [comment]
@illuzen Apologies for this - believe it's fixed in `4.1.5`. Thanks for reporting.
